youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/YouboraPlugin.java

```diff
@@ -274,16 +286,36 @@ private void handleTracksAvailable(PlayerEvent.TracksAvailable event, NPAWPlugin
     private static Options parseConfig(Object config) {
         if (config instanceof YouboraConfig) {
             houseHoldId = ((YouboraConfig) config).getHouseHoldId();
+            fastDataConfig = ((YouboraConfig) config).getFastDataConfig();
             return ((YouboraConfig) config).getYouboraOptions();
         } else if (config instanceof JsonObject) {
             YouboraConfig youboraConfig = new Gson().fromJson(((JsonObject) config), YouboraConfig.class);
             houseHoldId = youboraConfig.getHouseHoldId();
+            fastDataConfig = youboraConfig.getFastDataConfig();
             return youboraConfig.getYouboraOptions();
         } else if (config instanceof Bundle) {
-            Options options = new Options((Bundle) config);
+            Bundle configBundle = (Bundle) config;
+            Options options = new Options(configBundle);
             houseHoldId = ((Bundle) config).getString(KEY_HOUSEHOLD_ID);
+            fillFastDataConfig(configBundle);
+            return options;
+        } else if (config instanceof YouboraAdAdapterConfig) {
+            YouboraAdAdapterConfig adAdapterConfig = (YouboraAdAdapterConfig) config;
+            Options options = new Options(adAdapterConfig.getOptBundle());
+            houseHoldId = adAdapterConfig.getOptBundle().getString(KEY_HOUSEHOLD_ID);
+            customAdAdapter = adAdapterConfig.getCustomAdsAdapter();
+            fillFastDataConfig(adAdapterConfig.getOptBundle());
             return options;
         }
         return null;
     }
+
+    private static void fillFastDataConfig(Bundle configBundle) {
+        fastDataConfig = new ViewTransform.FastDataConfig();
```

Comment:
@ealfonso93 
what if we create FastDataConfig and no values
lets keep it null if nothing was sent